#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

histogram_policy_enum = gen.enum([gen.const("HUE", int_t, 0, "use hue only"),
                                  gen.const("HUE_AND_SATURATION", int_t, 1, "use hue and saturation")],
                                 "histogram policy")
gen.add("histogram_policy", int_t, 0, "histogram policy", 0, 0, 1,
        edit_method=histogram_policy_enum)
gen.add("histogram_index", int_t, 0, "index of histogram array to visualize", 0, 0, 100)
gen.add("histogram_scale", double_t, 0, "scale factor of histogram values", 1.0, 0.01, 1000.0)

exit(gen.generate(PACKAGE, PACKAGE, "ColorHistogramVisualizer"))
